tests/lib/rules/no-implicit-coercion.js

```diff
@@ -46,6 +51,15 @@ ruleTester.run("no-implicit-coercion", rule, {
         {code: "+foo.bar", errors: [{message: "use `Number(foo.bar)` instead.", type: "UnaryExpression"}]},
         {code: "1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
         {code: "foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*bar*1*foo", errors: [
+            {message: "use `Number(bar)` instead.", type: "BinaryExpression"},
+            {message: "use `Number(foo)` instead.", type: "BinaryExpression"}
+        ]},
+        {code: "1234*1*foo*bar", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
```

Comment:
Yes, in situations like `foo*1`. In situations like `1234*foo*1`, it's just unnecessary since the casting happens via `1234*` anyway. I'm saying I don't think this situation should be flagged by this rule because it's not casting.